components/navigation.tsx

```diff
@@ -137,41 +144,50 @@ const customNavDrop = {
     }
 }
 
-function OnHoverNav({ link, label, subLinks }: { link: string, label: string, subLinks: { subLink: string, label: string }[] | undefined }) {
+function handleMouseDown(e: React.MouseEvent, href: string, router: AppRouterInstance) {
```

Comment:
nitpicky comment: Should we maybe pass in the `push` function instead of the entire `router` instance?

I forgot the name of the principle/concept of not exposing information to a function that isn't needed by it, but in reality I just don't like the `AppRouterInstance` out of no real reason but aesthetics

Reply:
Yes I know what principle you mean (also forgot the name 🙈 ), while I do agree with your statement, it depends on the context. In this case it would be better to pass in the `AppRouterInstance` object because this way we are able to enforce that this function only works with this type.

If I would follow this principle here and just pass in the push function, I would have to add a void function callback parameter, e.g.:
```
function handleMouseDown(e: React.MouseEvent, href: string, push: () => void)
```

This allows the function caller to determine what the implementation of the push callback would be. This causes leaking implementation details of `handleMouseDown` and shifts the responsibility of how the push works (an implementation detail) to the function caller.